Dockerfile

```
@@ -0,0 +1,94 @@
+    uv sync --frozen
+
+FROM python:3.11-slim as runtime
+
+# Create non-root user with home directory (matching host UID/GID)
+RUN groupadd -g 1000 crawler && \
+    useradd -u 1000 -g 1000 -m -d /home/crawler crawler
+
+WORKDIR /app
+
+# Install system dependencies including Playwright browser dependencies in single layer
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Hadolint (2.12.0)

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)